Micronaut HTTP client fails with missing classpath dependency

### Expected Behavior

I expect that I can execute my HTTP client call with a new Micronaut Enabled service/application

### Actual Behaviour

An exception is thrown:

```
Error creating bean with name 'restService': Instantiation of supplied bean failed
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'restService': Instantiation of supplied bean failed
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.obtainFromSupplier(AbstractAutowireCapableBeanFactory.java:1263)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1200)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:569)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:529)
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:339)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:373)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:337)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireByName(AbstractAutowireCapableBeanFactory.java:1493)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1446)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireBeanProperties(AbstractAutowireCapableBeanFactory.java:405)
	at org.springframework.context.support.ApplicationContextAwareProcessor.invokeAwareInterfaces(ApplicationContextAwareProcessor.java:110)
	at org.springframework.context.support.ApplicationContextAwareProcessor.postProcessBeforeInitialization(ApplicationContextAwareProcessor.java:85)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.applyBeanPostProcessorsBeforeInitialization(AbstractAutowireCapableBeanFactory.java:429)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1818)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:419)
	at org.springframework.test.context.support.DependencyInjectionTestExecutionListener.injectDependencies(DependencyInjectionTestExecutionListener.java:157)
	at org.springframework.test.context.support.DependencyInjectionTestExecutionListener.prepareTestInstance(DependencyInjectionTestExecutionListener.java:111)
	at org.springframework.test.context.TestContextManager.prepareTestInstance(TestContextManager.java:260)
	at org.spockframework.spring.SpringTestContextManager.prepareTestInstance(SpringTestContextManager.java:56)
	at org.spockframework.spring.SpringInterceptor.interceptInitializerMethod(SpringInterceptor.java:46)
	at org.spockframework.runtime.extension.AbstractMethodInterceptor.intercept(AbstractMethodInterceptor.java:24)
	at org.spockframework.runtime.extension.MethodInvocation.proceed(MethodInvocation.java:101)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at org.spockframework.runtime.model.MethodInfo.invoke(MethodInfo.java:156)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
Caused by: io.micronaut.context.exceptions.BeanInstantiationException: Error instantiating bean of type  [io.netty.channel.EventLoopGroup]

Message: io/netty/channel/nio/NioIoHandler
Path Taken:
@i.m.c.a.Primary @j.i.Singleton i.n.c.EventLoopGroup i.m.h.n.c.DefaultEventLoopGroupRegistry.defaultEventLoopGroup#defaultEventLoopGroup(@Named("netty") ThreadFactory threadFactory)
	at io.micronaut.context.DefaultBeanContext.resolveByBeanFactory(DefaultBeanContext.java:2352)
	at io.micronaut.context.DefaultBeanContext.createRegistration(DefaultBeanContext.java:3150)
	at io.micronaut.context.SingletonScope.getOrCreate(SingletonScope.java:80)
	at io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:3000)
	at io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:2762)
	at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:1781)
	at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:855)
	at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:847)
	at io.micronaut.http.netty.channel.DefaultEventLoopGroupRegistry.getDefaultEventLoopGroup(DefaultEventLoopGroupRegistry.java:157)
	at io.micronaut.http.client.netty.DefaultNettyHttpClientRegistry.resolveEventLoopGroup(DefaultNettyHttpClientRegistry.java:480)
	at io.micronaut.http.client.netty.DefaultNettyHttpClientRegistry.clientBuilder(DefaultNettyHttpClientRegistry.java:466)
	at io.micronaut.http.client.netty.DefaultNettyHttpClientRegistry.lambda$getClient$1(DefaultNettyHttpClientRegistry.java:395)
	at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1708)
	at io.micronaut.http.client.netty.DefaultNettyHttpClientRegistry.getClient(DefaultNettyHttpClientRegistry.java:343)
	at io.micronaut.http.client.netty.DefaultNettyHttpClientRegistry.getClient(DefaultNettyHttpClientRegistry.java:208)
	at io.micronaut.http.client.netty.DefaultNettyHttpClientRegistry.resolveDefaultHttpClient(DefaultNettyHttpClientRegistry.java:509)
	at io.micronaut.http.client.netty.DefaultNettyHttpClientRegistry.httpClient(DefaultNettyHttpClientRegistry.java:277)
	at io.micronaut.context.AbstractInitializableBeanDefinition.instantiate(AbstractInitializableBeanDefinition.java:778)
	at io.micronaut.context.DefaultBeanContext.resolveByBeanFactory(DefaultBeanContext.java:2335)
	at io.micronaut.context.DefaultBeanContext.createRegistration(DefaultBeanContext.java:3150)
	at io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:3027)
	at io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:2762)
	at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:1781)
	at io.micronaut.context.AbstractBeanResolutionContext.getBean(AbstractBeanResolutionContext.java:210)
	at io.micronaut.context.AbstractInitializableBeanDefinition.resolveBean(AbstractInitializableBeanDefinition.java:2130)
	at io.micronaut.context.AbstractInitializableBeanDefinition.getBeanForMethodArgument(AbstractInitializableBeanDefinition.java:1159)
	at io.micronaut.context.DefaultBeanContext.resolveByBeanFactory(DefaultBeanContext.java:2337)
	at io.micronaut.context.DefaultBeanContext.createRegistration(DefaultBeanContext.java:3150)
	at io.micronaut.context.SingletonScope.getOrCreate(SingletonScope.java:80)
	at io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:3000)
	at io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:2969)
	at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:1747)
	at io.micronaut.spring.boot.starter.MicronautImportRegistrar.lambda$registerBeanWithContext$4(MicronautImportRegistrar.java:232)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.obtainInstanceFromSupplier(AbstractAutowireCapableBeanFactory.java:1297)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.obtainInstanceFromSupplier(DefaultListableBeanFactory.java:1033)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.obtainFromSupplier(AbstractAutowireCapableBeanFactory.java:1257)
	... 25 more
Caused by: java.lang.NoClassDefFoundError: io/netty/channel/nio/NioIoHandler
	at io.micronaut.http.netty.channel.NioEventLoopGroupFactory.createIoHandlerFactory(NioEventLoopGroupFactory.java:46)
	at io.micronaut.http.netty.channel.EventLoopGroupFactory.createIoHandlerFactory(EventLoopGroupFactory.java:75)
	at io.micronaut.http.netty.channel.DefaultEventLoopGroupFactory.createIoHandlerFactory(DefaultEventLoopGroupFactory.java:95)
	at io.micronaut.http.netty.channel.DefaultEventLoopGroupRegistry.createGroup(DefaultEventLoopGroupRegistry.java:103)
	at io.micronaut.http.netty.channel.DefaultEventLoopGroupRegistry.defaultEventLoopGroup(DefaultEventLoopGroupRegistry.java:151)
	at io.micronaut.context.DefaultBeanContext.resolveByBeanFactory(DefaultBeanContext.java:2337)
	... 60 more
Caused by: java.lang.ClassNotFoundException: io.netty.channel.nio.NioIoHandler
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526)
	... 66 more


Error instantiating bean of type  [io.netty.channel.EventLoopGroup]

Message: io/netty/channel/nio/NioIoHandler
Path Taken:
@i.m.c.a.Primary @j.i.Singleton i.n.c.EventLoopGroup i.m.h.n.c.DefaultEventLoopGroupRegistry.defaultEventLoopGroup#defaultEventLoopGroup(@Named("netty") ThreadFactory threadFactory)
io.micronaut.context.exceptions.BeanInstantiationException: Error instantiating bean of type  [io.netty.channel.EventLoopGroup]

Message: io/netty/channel/nio/NioIoHandler
Path Taken:
@i.m.c.a.Primary @j.i.Singleton i.n.c.EventLoopGroup i.m.h.n.c.DefaultEventLoopGroupRegistry.defaultEventLoopGroup#defaultEventLoopGroup(@Named("netty") ThreadFactory threadFactory)
	at app//io.micronaut.context.DefaultBeanContext.resolveByBeanFactory(DefaultBeanContext.java:2352)
	at app//io.micronaut.context.DefaultBeanContext.createRegistration(DefaultBeanContext.java:3150)
	at app//io.micronaut.context.SingletonScope.getOrCreate(SingletonScope.java:80)
	at app//io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:3000)
	at app//io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:2762)
	at app//io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:1781)
	at app//io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:855)
	at app//io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:847)
	at app//io.micronaut.http.netty.channel.DefaultEventLoopGroupRegistry.getDefaultEventLoopGroup(DefaultEventLoopGroupRegistry.java:157)
	at app//io.micronaut.http.client.netty.DefaultNettyHttpClientRegistry.resolveEventLoopGroup(DefaultNettyHttpClientRegistry.java:480)
	at app//io.micronaut.http.client.netty.DefaultNettyHttpClientRegistry.clientBuilder(DefaultNettyHttpClientRegistry.java:466)
	at app//io.micronaut.http.client.netty.DefaultNettyHttpClientRegistry.lambda$getClient$1(DefaultNettyHttpClientRegistry.java:395)
	at java.base@21.0.5/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1708)
	at app//io.micronaut.http.client.netty.DefaultNettyHttpClientRegistry.getClient(DefaultNettyHttpClientRegistry.java:343)
	at app//io.micronaut.http.client.netty.DefaultNettyHttpClientRegistry.getClient(DefaultNettyHttpClientRegistry.java:208)
	at app//io.micronaut.http.client.netty.DefaultNettyHttpClientRegistry.resolveDefaultHttpClient(DefaultNettyHttpClientRegistry.java:509)
	at app//io.micronaut.http.client.netty.DefaultNettyHttpClientRegistry.httpClient(DefaultNettyHttpClientRegistry.java:277)
	at app//io.micronaut.context.AbstractInitializableBeanDefinition.instantiate(AbstractInitializableBeanDefinition.java:778)
	at app//io.micronaut.context.DefaultBeanContext.resolveByBeanFactory(DefaultBeanContext.java:2335)
	at app//io.micronaut.context.DefaultBeanContext.createRegistration(DefaultBeanContext.java:3150)
	at app//io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:3027)
	at app//io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:2762)
	at app//io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:1781)
	at app//io.micronaut.context.AbstractBeanResolutionContext.getBean(AbstractBeanResolutionContext.java:210)
	at app//io.micronaut.context.AbstractInitializableBeanDefinition.resolveBean(AbstractInitializableBeanDefinition.java:2130)
	at app//io.micronaut.context.AbstractInitializableBeanDefinition.getBeanForMethodArgument(AbstractInitializableBeanDefinition.java:1159)
	at app//io.micronaut.context.DefaultBeanContext.resolveByBeanFactory(DefaultBeanContext.java:2337)
	at app//io.micronaut.context.DefaultBeanContext.createRegistration(DefaultBeanContext.java:3150)
	at app//io.micronaut.context.SingletonScope.getOrCreate(SingletonScope.java:80)
	at app//io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:3000)
	at app//io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:2969)
	at app//io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:1747)
	at app//io.micronaut.spring.boot.starter.MicronautImportRegistrar.lambda$registerBeanWithContext$4(MicronautImportRegistrar.java:232)
	at app//org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.obtainInstanceFromSupplier(AbstractAutowireCapableBeanFactory.java:1297)
	at app//org.springframework.beans.factory.support.DefaultListableBeanFactory.obtainInstanceFromSupplier(DefaultListableBeanFactory.java:1033)
	at app//org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.obtainFromSupplier(AbstractAutowireCapableBeanFactory.java:1257)
	at app//org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1200)
	at app//org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:569)
	at app//org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:529)
	at app//org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:339)
	at app//org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:373)
	at app//org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:337)
	at app//org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202)
	at app//org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireByName(AbstractAutowireCapableBeanFactory.java:1493)
	at app//org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1446)
	at app//org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireBeanProperties(AbstractAutowireCapableBeanFactory.java:405)
	at app//org.springframework.context.support.ApplicationContextAwareProcessor.invokeAwareInterfaces(ApplicationContextAwareProcessor.java:110)
	at app//org.springframework.context.support.ApplicationContextAwareProcessor.postProcessBeforeInitialization(ApplicationContextAwareProcessor.java:85)
	at app//org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.applyBeanPostProcessorsBeforeInitialization(AbstractAutowireCapableBeanFactory.java:429)
	at app//org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1818)
	at app//org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:419)
	at app//org.springframework.test.context.support.DependencyInjectionTestExecutionListener.injectDependencies(DependencyInjectionTestExecutionListener.java:157)
	at app//org.springframework.test.context.support.DependencyInjectionTestExecutionListener.prepareTestInstance(DependencyInjectionTestExecutionListener.java:111)
	at app//org.springframework.test.context.TestContextManager.prepareTestInstance(TestContextManager.java:260)
	at app//org.spockframework.spring.SpringTestContextManager.prepareTestInstance(SpringTestContextManager.java:56)
	at app//org.spockframework.spring.SpringInterceptor.interceptInitializerMethod(SpringInterceptor.java:46)
	at app//org.spockframework.runtime.extension.AbstractMethodInterceptor.intercept(AbstractMethodInterceptor.java:24)
	at app//org.spockframework.runtime.extension.MethodInvocation.proceed(MethodInvocation.java:101)
	at java.base@21.0.5/java.util.ArrayList.forEach(ArrayList.java:1596)
	at app//org.spockframework.runtime.model.MethodInfo.invoke(MethodInfo.java:156)
	at java.base@21.0.5/java.util.ArrayList.forEach(ArrayList.java:1596)
Caused by: java.lang.NoClassDefFoundError: io/netty/channel/nio/NioIoHandler
	at io.micronaut.http.netty.channel.NioEventLoopGroupFactory.createIoHandlerFactory(NioEventLoopGroupFactory.java:46)
	at io.micronaut.http.netty.channel.EventLoopGroupFactory.createIoHandlerFactory(EventLoopGroupFactory.java:75)
	at io.micronaut.http.netty.channel.DefaultEventLoopGroupFactory.createIoHandlerFactory(DefaultEventLoopGroupFactory.java:95)
	at io.micronaut.http.netty.channel.DefaultEventLoopGroupRegistry.createGroup(DefaultEventLoopGroupRegistry.java:103)
	at io.micronaut.http.netty.channel.DefaultEventLoopGroupRegistry.defaultEventLoopGroup(DefaultEventLoopGroupRegistry.java:151)
	at io.micronaut.context.DefaultBeanContext.resolveByBeanFactory(DefaultBeanContext.java:2337)
	... 60 more
Caused by: java.lang.ClassNotFoundException: io.netty.channel.nio.NioIoHandler
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526)
	... 66 more
```


### Steps To Reproduce

Run the integration tests of 

https://github.com/sbglasius/grails7-micronaut-bug



### Environment Information

- OS: Ubuntu 24.4
- JDK: 21.0.5 Liberica

### Example Application

https://github.com/sbglasius/grails7-micronaut-bug

### Version

7.0.0-RC2

I encountered a very similar bug in a Grails 7.0.0 application created using Grails Forge with the Micronaut HTTP Client additional feature.

The `io.micronaut.http.client.HttpClient.create(@Nullable URL url)` method threw an exception in my integration test:
```
java.lang.NoClassDefFoundError: io/netty/channel/MultiThreadIoEventLoopGroup
    at io.micronaut.http.client.netty.ConnectionManager.createEventLoopGroup(ConnectionManager.java:277)
    at io.micronaut.http.client.netty.ConnectionManager.<init>(ConnectionManager.java:234)
    at io.micronaut.http.client.netty.DefaultHttpClient.<init>(DefaultHttpClient.java:424)
    at io.micronaut.http.client.netty.DefaultHttpClientBuilder.build(DefaultHttpClientBuilder.java:283)
    at io.micronaut.http.client.netty.NettyHttpClientFactory.createNettyClient(NettyHttpClientFactory.java:141)
    at io.micronaut.http.client.netty.NettyHttpClientFactory.createNettyClient(NettyHttpClientFactory.java:126)
    at io.micronaut.http.client.netty.NettyHttpClientFactory.createClient(NettyHttpClientFactory.java:57)
    at io.micronaut.http.client.HttpClient.create(HttpClient.java:251)
    at com.example.HelloControllerIntegrationSpec.$tt__init(HelloControllerIntegrationSpec.groovy:24)
    at com.example.HelloControllerIntegrationSpec.init_closure1(HelloControllerIntegrationSpec.groovy)
    at groovy.lang.Closure.call(Closure.java:433)
    at groovy.lang.Closure.call(Closure.java:422)
    at grails.gorm.transactions.GrailsTransactionTemplate$2.doInTransaction(GrailsTransactionTemplate.groovy:98)
    at org.springframework.transaction.support.TransactionTemplate.execute(TransactionTemplate.java:140)
    at grails.gorm.transactions.GrailsTransactionTemplate.execute(GrailsTransactionTemplate.groovy:95)
    at org.spockframework.util.ReflectionUtil.invokeMethod(ReflectionUtil.java:187)
    at org.spockframework.runtime.model.MethodInfo.lambda$new$0(MethodInfo.java:49)
    at org.spockframework.runtime.model.MethodInfo.invoke(MethodInfo.java:156)
    at org.spockframework.runtime.extension.MethodInvocation.proceed(MethodInvocation.java:102)
    at org.grails.testing.spock.RunOnceInterceptor.interceptSetupMethod(RunOnceInterceptor.groovy:36)
    at org.spockframework.runtime.extension.AbstractMethodInterceptor.intercept(AbstractMethodInterceptor.java:30)
    at org.spockframework.runtime.extension.MethodInvocation.proceed(MethodInvocation.java:101)
    at org.spockframework.runtime.model.MethodInfo.invoke(MethodInfo.java:156)
    at org.spockframework.runtime.extension.MethodInvocation.proceed(MethodInvocation.java:102)
    at org.spockframework.spring.SpringInterceptor.interceptSetupMethod(SpringInterceptor.java:55)
    at org.spockframework.runtime.extension.AbstractMethodInterceptor.intercept(AbstractMethodInterceptor.java:30)
    at org.spockframework.runtime.extension.MethodInvocation.proceed(MethodInvocation.java:101)
    at org.spockframework.runtime.model.MethodInfo.invoke(MethodInfo.java:156)
    at org.spockframework.runtime.model.MethodInfo.invoke(MethodInfo.java:156)
    at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
    at org.spockframework.runtime.model.MethodInfo.invoke(MethodInfo.java:156)
    at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
Caused by: java.lang.ClassNotFoundException: io.netty.channel.MultiThreadIoEventLoopGroup
    at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
    at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
    at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)
    ... 32 more
```

This is because Micronaut platform version (4.9.2) expects Netty 4.2.x, but Spring Boot 3.5.6 (from Grails BOM) forces Netty 4.1.x. And the `io.netty.channel.MultiThreadIoEventLoopGroup` class was introduced in Netty 4.2.x.

I created a sample application to demonstrate the issue:
https://github.com/tylervz/grails700_micronaut_http_bug

Following the suggestion of an LLM (because I didn't see any messages about `java.lang.NoClassDefFoundError: io/netty/channel/MultiThreadIoEventLoopGroup` in the [Grails Slack channel](https://www.linen.dev/s/grails/c/questions)), I was able to resolve the issue by downgrading the Micronaut platform version, specified in `gradle.properties`, from 4.9.2 to 4.7.6 which is compatible with Netty 4.1.x. Micronaut platforms 4.8+ moved to Netty 4.2.x.

I'd like to help fix this bug, my work schedule permitting, but I don't know where in the Grails code base the `micronautPlatformVersion` value for `gradle.properties` is defined for `grails-forge-cli create-app` commands that include the `micronaut-http-client` feature such as this: `grails-forge-cli create-app --servlet=tomcat --jdk=17 --gorm=hibernate --test=spock --features=micronaut-http-client com.example.sample`

I'm thinking the `micronautPlatformVersion` should be lowered to 4.7.x and an integration test should be added to ensure the `micronaut-http-client` runs properly when upgrades are made to the `micronautPlatformVersion` or the Grails BOM.

--- a/grails-forge/grails-forge-core/src/main/java/org/grails/forge/feature/micronaut/MicronautHttpClient.java
+++ b/grails-forge/grails-forge-core/src/main/java/org/grails/forge/feature/micronaut/MicronautHttpClient.java
@@ -91,7 +91,7 @@ public class MicronautHttpClient implements Feature {
 
         generatorContext.addDependency(Dependency.builder()
                 .groupId("io.micronaut")
-                .artifactId("micronaut-http-client")
+                .artifactId("micronaut-http-client-jdk") // Use jdk version as Spring and Micronaut has different Netty versions https://github.com/apache/grails-core/issues/15149
                 .implementation());
 
         // micronaut-http-client no longer provides the jackson implementation

